include/mvd/bits/mvd3_misc.hpp

```diff
@@ -206,6 +210,21 @@ inline std::vector<std::string> MVD3File::getSynapseClass(const Range & range) c
 }
 
 
+inline std::vector<TSV::TSVInfo> MVD3File::getTSVInfo(const Range & range ) const{
+    std::vector<std::string> etypes = getEtypes(range);
+    std::vector<boost::int32_t> layers = getLayers(range);
+    std::vector<std::string> morphologies = getMorphologies(range);
+    std::vector<TSV::TSVInfo> TSVInfos;
+    for (auto mecombo_tupple : boost::combine(etypes, layers, morphologies)) {
+        std::string etype, morphology;
+        boost::int32_t layer;
+        boost::tie(etype, layer, morphology) = mecombo_tupple;
+        TSVInfos.push_back(_tsv_file->getTSVInfo(etype + "_" + std::to_string(layer) + "_" + morphology, morphology));
```

Comment:
What happens if `_tsv_file == nullptr`? I think there should be a guard and an exception raised…

Reply:
Good catch! :+1: